### PR TITLE
Move cert-manager/istio-csr as a member of cert-manager

### PR DESCRIPTION
**Pull request by JoshVanL**
Signed-off-by: joshvanl <vleeuwenjoshua@gmail.com>

Adds the `carotation` test to istio-csr.

Now that istio-csr has moved its default head branch to main, this should be merged straight away.

/assign @irbekrm 

```release-note
NONE
```

### CODE REVIEW

File: config/jobs/cert-manager/istio-csr/cert-manager-istio-csr-presubmits.yaml [hunk 1]
```diff
@@ -19,17 +19,61 @@ presubmits:
             cpu: 1
             memory: 1Gi
 
+  - name: pull-cert-manager-istio-csr-ca-rotation
+    context: pull-cert-manager-istio-csr-ca-rotation
+    agent: kubernetes
```
Comment (irbekrm) [inline config/jobs/cert-manager/istio-csr/cert-manager-istio-csr-presubmits.yaml:24]: Nit: add branches: [`^main$`]

**Review by irbekrm — CHANGES_REQUESTED** [review]
Looks good to me, I've added a nit to specify a branch for the new job, it looks to me that if unspecified it runs against all branches https://github.com/kubernetes/test-infra/blob/51a9929c946b/prow/config/jobs.go#L225

File: config/jobs/cert-manager/istio-csr/cert-manager-istio-csr-presubmits.yaml (hunk 1 above)
Comment (JoshVanL) [reply config/jobs/cert-manager/istio-csr/cert-manager-istio-csr-presubmits.yaml:26]: @irbekrm done

**Review by irbekrm — APPROVED** [review]
/lgtm

**Pull request merged**